File: day_three/part_one.py
def tree_counter(file: str) -> int:
    with open(file, "r") as f:
        total = 0
        lines = [line.replace("\n", "") for line in f.readlines()]
        x, y = 0, 0
        x += 3
        x %= len(lines[0])
        y += 1
        while y < len(lines):
            if lines[y][x] == "#":
                total += 1
            x += 3
            x %= len(lines[0])
            y += 1

    return total


if __name__ == "__main__":
    print(tree_counter("./blob.txt"))

File: day_three/go/part_one/part_one.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
)

func treeCounter() int {

	f, err := os.Open("../blob.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer f.Close()

	scanner := bufio.NewScanner(f)
	total := 0
	var trees []string
	for scanner.Scan() {
		text := scanner.Text()
		trees = append(trees, text)
	}

	x := 0
	y := 0
	x += 3
	x %= len(trees[0])
	y ++
	for y < len(trees) {
		if trees[y][x] == '#' {
			total ++
		}
		x += 3
		x %= len(trees[0])
		y ++
	}

return total
}

func main() {
	fmt.Println(treeCounter())
}
